python/pyspark/sql/connect/session.py

```diff
@@ -456,7 +455,7 @@ def register_udf(self, function: Any, return_type: Union[str, DataType]) -> str:
     @staticmethod
     def _start_connect_server(master: str, opts: Dict[str, Any]) -> None:
         """
-        Starts the Spark Connect server given the master.
+        Starts the Spark Connect server given the master (thread-unsafe).
```

Comment:
Oh, is it okay?

Reply:
Ah, yeah. So this is invoked with the lock: https://github.com/apache/spark/blob/31be205bd6ac8e6eb77fc9b876298e0848dbf929/python/pyspark/sql/session.py#L426-L439